src/components/LoginRegister/RegisterInput.js

```diff
@@ -0,0 +1,75 @@
+import React, { useState } from 'react';
+import { useApolloClient } from '@apollo/client';
+import { useDispatch } from 'react-redux';
+import { Flex, Input, Button, Heading, VStack } from '@chakra-ui/react';
+import { register } from '../../data/actions/auth';
+import { Redirect } from 'react-router-dom';
+
+const RegisterInput = () => {
+  const client = useApolloClient();
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [submitState, setSubmitState] = useState('');
+  const dispatch = useDispatch();
+
+  const handleClick = async e => {
+    e.preventDefault();
+    dispatch(register(firstName, lastName, email, password, client)).then(
+      registered => {
+        if (!registered) {
+          setSubmitState('FAIL');
+        } else {
+          // Redirect to home page after registering
+          return <Redirect to={{ pathname: '/home' }} />;
+        }
+      },
+      error => {
+        setSubmitState('FAIL');
+      },
+    );
+  };
+
+  const onChangeFirstName = e => {
+    setFirstName(e.target.value);
+  };
+
+  const onChangeLastName = e => {
+    setLastName(e.target.value);
+  };
+
+  const onChangeEmail = e => {
+    setEmail(e.target.value);
+  };
+
+  const onChangePass = e => {
+    setPassword(e.target.value);
+  };
+
+  return (
+    <form onSubmit={e => handleClick(e)}>
+      <VStack spacing="24px">
+        <Flex w="100%" justify="left">
+          <Heading as="h1">Create an account</Heading>
+        </Flex>
+        <Input name="firstName" placeholder="First Name" value={firstName} onChange={onChangeFirstName} />
+        <Input name="lastName" placeholder="Last Name" value={lastName} onChange={onChangeLastName} />
+        <Input name="email" placeholder="Email" value={email} onChange={onChangeEmail} />
```

Comment:
add `type="email"` here and you actually get some default form validation! It doesn't validate "not empty" but it does validate you enter a valid email format if you do start typing something :O 

![image](https://user-images.githubusercontent.com/26701004/109525964-0e584300-7a80-11eb-8dde-e7ee71890c6d.png)

```suggestion
        <Input type="email" name="email" placeholder="Email" value={email} onChange={onChangeEmail} />
```

Reply:
Done!